source/system/structs/HashMap.ooc

```diff
@@ -6,85 +6,50 @@
  * of the MIT license.  See the LICENSE file for details.
  */
 
-HashEntry: cover {
-	key: Pointer
-	value: Pointer
-	next: This* = null
-	init: func@ { this init(null, null) }
-	init: func@ ~keyVal (=key, =value)
-	free: func {
-		if (this key != null)
-			memfree(this key)
-		if (this value != null)
-			memfree(this value)
-		if (this next != null) {
-			temp := this next@
-			temp free()
-			memfree(this next)
+_HashEntry: class <K, V> {
+	_key: __onheap__ K
+	_value: __onheap__ V
+	_next: This<K, V> = null
+	init: func (=_key, =_value)
+	free: override func {
+		memfree(this _key)
+		memfree(this _value)
+		if (this _next != null) {
+			this _next free()
+			this _next = null
 		}
+		super()
 	}
 }
 
-intHash: func <K> (key: K) -> SizeT { (key as Int) as SizeT }
-pointerHash: func <K> (key: K) -> SizeT { (key as Pointer) as SizeT }
-
-/**
-	Port of Austin Appleby's Murmur Hash implementation
-	http://code.google.com/p/smhasher/
-*/
-murmurHash: func <K> (keyTagazok: K) -> SizeT {
-	seed: SizeT = 1
-
-	len := K size
-	m = 0x5bd1e995: SizeT
-	r = 24: SSizeT
-	l := len
-
-	h : SizeT = seed ^ len
-	data := (keyTagazok&) as Byte*
-
-	while (true) {
+// Austin Appleby's Murmur Hash implementation: http://code.google.com/p/smhasher/
+_murmurHash: func <K> (key: K) -> Int {
+	(len, m, r, l, h, t) := (K size, 0x5bd1e995 as SizeT, 24 as SizeT, len, (1 as SizeT) ^ len as SizeT, 0)
+	data := (key&) as Byte*
+	while (len >= 4) {
```

Comment:
@davidpiuva I changed this so that it works for types that are < 4 bytes (like `Char` in one of the tests).